Add support for ECS External launch type

cuttius opened the pull request:
## Summary & Motivation
We are using Dagster in AWS using the ECS launcher. For certain tasks we are looking to move to Dagster the software has to be run on machines that are in our data center. We would like to be able to use the Dagster with ECS using ECS Anywhere, but found when trying to launch tasks with the External launch type Dagster would fail to start tasks (attached error in screenshot). The error was created during the RunTask boto3 call since the awsvpc networkMode is only valid if the launch type is "EC2" or "FARGATE". This change adds a check for the launch type before applying the aws vpc network configuration settings.

## How I Tested These Changes
I changed the dagster-aws module to be pip installed from my git repository and was able to run tasks on External instances.

<img width="1218" alt="networkMode_error" src="https://github.com/dagster-io/dagster/assets/5329455/ba2e5de6-519c-4941-a7b1-d48b6e3404e4">

gibsondan commented on python_modules/libraries/dagster-aws/dagster_aws/ecs/tasks.py:
```diff
@@ -375,20 +375,19 @@ def get_task_kwargs_from_current_task(
         run_task_kwargs["capacityProviderStrategy"] = task.get("capacityProviderStrategy")
 
+    if run_task_kwargs["launchType"] != "EXTERNAL":
+        aws_vpc_config = {
+            "subnets": subnets,
+            "assignPublicIp": "ENABLED" if public_ip else "DISABLED",
+            "securityGroups": security_groups,
```
we can probably move the places where we calculate subnets/public_ip/security_groups inside this if block as well, since it's now the only place where they are used

cuttius replied:
Yeah, makes sense to setup the vpc config after the check.

gibsondan reviewed (changes requested):
thanks for sending this out! this seems reasonable to me - I think the main missing piece here is a test in this file that includes an EXTERNAL launch type: https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-aws/dagster_aws_tests/ecs_tests/launcher_tests/test_launching.py - once that's in this looks great to me

cuttius commented:
I added the test case for EXTERNAL launch type. The EXTERNAL launch type also requires bridge networkMode. I added a new task definition to test these. Let me know if you prefer a different way to test it. Thanks for the feedback and review.

gibsondan reviewed (approved):
looks great, thanks!
